project/BuildInfo.scala

```diff
@@ -26,13 +27,16 @@ object BuildInfo {
     file: File,
     log: ManagedLogger,
     ensoVersion: String,
+    defaultDevEnsoVersion: String,
     scalacVersion: String,
     graalVersion: String,
     currentEdition: String,
     stdLibVersion: String
   ): Seq[File] = {
     val gitInfo   = getGitInformation(log).getOrElse(fallbackGitInformation)
     val isRelease = isReleaseMode
+    val isInternalDevelopmentBuild: Boolean =
+      ensoVersion == defaultDevEnsoVersion
```

Comment:
Also, we already have `isRelease`. Why not specify dev build as an inversion of release, like `!isRelease`?

Reply:
We may want a development build not in the 'internal' mode - for example to test how it behaves with a changed version string, but not in release mode.
If I understand correctly, our usual CI will actually run with `isInternalDevelopmentBuild == false` as it will override the version (and so for example the CI artifacts will be not internal-dev but not release either).

Reply:
This flag was actually used to instrument some badly behaving tests. I figured a different and better way to tackle this, so I think it can actually be removed.